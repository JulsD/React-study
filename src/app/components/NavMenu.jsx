import React from 'react';

import btn from './../../styles/buttons.css';
import styles from './../../styles/nav-menu.css';

class NavMenu extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const menu = this.props.menu;
    const menuList = menu.map((menuItem) =>
      <li key={menuItem.toString()}>
        <a href="#search" className= {`${btn.btn} ${btn.btnGray}`}>{menuItem}</a>
      </li>
    );
    return (
      <ul className = {styles.root}>
        {menuList}
      </ul>
    );
  }

}

export default NavMenu;
