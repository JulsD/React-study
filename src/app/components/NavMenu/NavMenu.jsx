import React from 'react';
import { Link } from 'react-router-dom'


import btn from './../../../styles/buttons.css';
import styles from './nav-menu.css';

class NavMenu extends React.Component {

  constructor(props) {
    super(props);

  }

  render() {
    const menu = this.props.menu;
    const menuList = menu.map((menuItem) =>
      <li key={menuItem.toString()}>
        <Link to='/' className= {`${btn.btn} ${btn.btnGray}`}>{menuItem}</Link>
      </li>
    );
    return (
      <ul className = {styles.root}>
        {menuList}
      </ul>
    );
  }

}

export default NavMenu;
