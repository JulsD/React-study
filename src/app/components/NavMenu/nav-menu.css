.root {
  list-style: none;
  padding: 0;
  text-align: right;
}
.root li {
  margin-left: 20px;
  display: inline-block;
}
.root li a {
  text-decoration: none;
  cursor: pointer;
}