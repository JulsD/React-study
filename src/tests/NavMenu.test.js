import React from 'react';
import NavMenu from '../app/components/NavMenu';

test('NavMenu should render a <ul>', () => {
  const wrapper = shallow(
    <NavMenu menu={['Search']}/>
  );
  expect(wrapper).toMatchSnapshot();
});
