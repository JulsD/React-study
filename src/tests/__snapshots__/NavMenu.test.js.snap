// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu should render a <ul> 1`] = `
<ul
  className="root"
>
  <li
    key="Search"
  >
    <Button
      kind="default"
      to="/"
      type="link"
    >
      Search
    </Button>
  </li>
</ul>
`;
